ingest/processors/token_transfer/token_transfer_processor.go

```diff
@@ -0,0 +1,736 @@
+		toAddress = protoAddressFromClaimableBalanceId(*to.claimableBalanceId)
+	}
+
+	protoAsset := assetProto.NewProtoAsset(asset)
+
+	var event *TokenTransferEvent
+	if isMintEvent {
+		// Mint event
+		event = NewMintEvent(meta, toAddress, amt, protoAsset)
+	} else if isBurnEvent {
+		// Burn event
+		event = NewBurnEvent(meta, fromAddress, amt, protoAsset)
+	} else {
+		event = NewTransferEvent(meta, fromAddress, toAddress, amt, protoAsset)
+	}
+	return event
+}
+
+func paymentEvents(tx ingest.LedgerTransaction, opIndex uint32, op xdr.Operation) ([]*TokenTransferEvent, error) {
+	paymentOp := op.Body.MustPaymentOp()
+	opSrcAcc := operationSourceAccount(tx, op)
+	destAcc := paymentOp.Destination
+	amt := amount.String(paymentOp.Amount)
+	meta := NewEventMeta(tx, &opIndex, nil)
+
+	from, to := addressWrapper{account: &opSrcAcc}, addressWrapper{account: &destAcc}
+	event := mintOrBurnOrTransferEvent(paymentOp.Asset, from, to, amt, meta)
+	return []*TokenTransferEvent{event}, nil
+}
+
+func createClaimableBalanceEvents(tx ingest.LedgerTransaction, opIndex uint32, op xdr.Operation, result xdr.OperationResult) ([]*TokenTransferEvent, error) {
+	createCbOp := op.Body.MustCreateClaimableBalanceOp()
+	createCbResult := result.Tr.MustCreateClaimableBalanceResult()
+	opSrcAcc := operationSourceAccount(tx, op)
+	meta := NewEventMeta(tx, &opIndex, nil)
+	claimableBalanceId := createCbResult.MustBalanceId()
+
+	from, to := addressWrapper{account: &opSrcAcc}, addressWrapper{claimableBalanceId: &claimableBalanceId}
+	event := mintOrBurnOrTransferEvent(createCbOp.Asset, from, to, amount.String(createCbOp.Amount), meta)
+	return []*TokenTransferEvent{event}, nil
+}
+
+func generateClaimableBalanceIdFromLiquidityPoolId(lpEntry liquidityPoolEntryDelta, tx ingest.LedgerTransaction, txSrcAccount xdr.AccountId, opIndex uint32) ([]xdr.ClaimableBalanceId, error) {
+	var generatedClaimableBalanceIds []xdr.ClaimableBalanceId
+	lpId := lpEntry.liquidityPoolId
+	seqNum := xdr.SequenceNumber(tx.Envelope.SeqNum())
+
+	for _, asset := range []xdr.Asset{lpEntry.assetA, lpEntry.assetB} {
+		cbId, err := converters.ConvertLiquidityPoolIdToClaimableBalanceId(lpId, asset, seqNum, txSrcAccount, opIndex)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to generate claimable balance id from LiquidityPoolId: %v, for asset: %v", lpIdToStrkey(lpId), asset.String())
+		}
+		generatedClaimableBalanceIds = append(generatedClaimableBalanceIds, cbId)
+	}
+	return generatedClaimableBalanceIds, nil
+}
+
+// This operation is used to only find CB entries that are either created or deleted, not updated
+func getClaimableBalanceEntriesFromOperationChanges(changeType xdr.LedgerEntryChangeType, tx ingest.LedgerTransaction, opIndex uint32) ([]xdr.ClaimableBalanceEntry, error) {
+	if !(changeType == xdr.LedgerEntryChangeTypeLedgerEntryRemoved || changeType == xdr.LedgerEntryChangeTypeLedgerEntryCreated) {
+		return nil, fmt.Errorf("changeType: %v, not allowed", changeType)
+	}
```

Comment:
I don't think this error is necessary. The function will obtain all cb changes occurring from an operation filtered by a change type. The fact that the function happens to be called only for removed or created changes is something determined by the caller

Reply:
you could simplify the if statement condition to:

```suggestion
	if changeType == xdr.LedgerEntryChangeTypeLedgerEntryUpdated {
		return nil, fmt.Errorf("LEDGER_ENTRY_UPDATED is not a valid filter")
	}
```